File: VB/FilterDeserializer/FilterDeserializer.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.Utils.Serializing
Imports DevExpress.Utils.Serializing.Helpers
Imports System.IO

Namespace FilterDeserializer
	Public Class FilterDeserializer
		Inherits XmlXtraSerializer
		Private Shared instance_Renamed As FilterDeserializer = Nothing
		Public Shared ReadOnly Property Instance() As FilterDeserializer
			Get
				If instance_Renamed Is Nothing Then
					instance_Renamed = New FilterDeserializer()
				End If
				Return instance_Renamed
			End Get
		End Property
		Public Shared Sub LoadFilter(ByVal view As ColumnView, ByVal file As String)
			Dim collection As IXtraPropertyCollection = Instance.Deserialize(file, "View", Nothing)
			Instance.DeserializeObject(view, GetFilterCollection(view, collection), view.OptionsLayout)
		End Sub
		Private Shared Function GetFilterCollection(ByVal view As ColumnView, ByVal collection As IXtraPropertyCollection) As IXtraPropertyCollection
			Dim filterNames As String = "|ActiveFilterEnabled|ActiveFilterString|MRUFilters|"
			Dim ret As IXtraPropertyCollection
			Using ms As Stream = New MemoryStream()
				view.SaveLayoutToStream(ms, view.OptionsLayout)
				ms.Position = 0
				ret = Instance.Deserialize(ms, "View", Nothing)
				For Each pi As XtraPropertyInfo In collection
					If filterNames.Contains(String.Format("|{0}|", pi.Name)) Then
						ret.Add(pi)
					End If
				Next pi
			End Using
			Return ret
		End Function
	End Class
End Namespace

File: CS/FilterDeserializer/FilterDeserializer.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.Utils.Serializing;
using DevExpress.Utils.Serializing.Helpers;
using System.IO;

namespace FilterDeserializer {
    public class FilterDeserializer : XmlXtraSerializer {
        static FilterDeserializer instance = null;
        public static FilterDeserializer Instance {
            get {
                if(instance == null)
                    instance = new FilterDeserializer();
                return instance;
            }
        }
        public static void LoadFilter(ColumnView view, string file) {
            IXtraPropertyCollection collection = Instance.Deserialize(file, "View", null);
            Instance.DeserializeObject(view, GetFilterCollection(view, collection), view.OptionsLayout);
        }
        static IXtraPropertyCollection GetFilterCollection(ColumnView view, IXtraPropertyCollection collection) {
            string filterNames = "|ActiveFilterEnabled|ActiveFilterString|MRUFilters|";
            IXtraPropertyCollection ret;
            using(Stream ms = new MemoryStream()) {
                view.SaveLayoutToStream(ms, view.OptionsLayout);
                ms.Position = 0;
                ret = Instance.Deserialize(ms, "View", null);
                foreach(XtraPropertyInfo pi in collection)
                    if(filterNames.Contains(string.Format("|{0}|", pi.Name)))
                        ret.Add(pi);
            }
            return ret;
        }
    }
}
